### PR TITLE
Support new features

### PR DESCRIPTION
**Pull request by joker1007**
Support new features. (main purpose is Fargate support)

- Support placement_constraints and strategies
- Support Task level CPU, Memory constraints
- Support require_compatibilities, launch_type for Fargate
- Support network_configuration for awsvpc network mode
- Enable to delete unnecessary service by ecs_deploy

### CODE REVIEW

File: lib/ecs_deploy/capistrano.rb [hunk 1]
```diff
@@ -91,8 +94,14 @@
             task_definition_name: service[:task_definition_name],
             load_balancers: service[:load_balancers],
             desired_count: service[:desired_count],
+            launch_type: service[:launch_type],
+            health_check_grace_period_seconds: service[:health_check_grace_period_seconds],
+            delete: service[:delete],
           }
           service_options[:deployment_configuration] = service[:deployment_configuration] if service[:deployment_configuration]
+          service_options[:deployment_configuration] = service[:deployment_configuration] if service[:deployment_configuration]
```
Comment (abicky) [inline lib/ecs_deploy/capistrano.rb]: [nits] duplicate line

File: lib/ecs_deploy/service.rb [hunk 2]
```diff
@@ -5,26 +5,39 @@ class Service
...
 
     def initialize(
       cluster:, service_name:, task_definition_name: nil, revision: nil,
       load_balancers: nil,
       desired_count: nil, deployment_configuration: {maximum_percent: 200, minimum_healthy_percent: 100},
-      region: nil
+      launch_type: nil,
+      placement_constraints: [],
+      placement_strategy: [],
+      network_configuration: nil,
```
Comment (abicky) [inline lib/ecs_deploy/service.rb]: [nits] This argument seems to be never specified from capistrano tasks.

**Comment by joker1007** [comment]
@abicky 
I've fixed.

**Comment by abicky** [comment]
LGTM!

**Pull request merged**